lunavl/sdk/indexes/builder.py

```diff
@@ -0,0 +1,116 @@
+"""
+Module contains core index builder.
+"""
+from pathlib import Path
+from typing import Union
+
+from FaceEngine import PyIFaceEngine
+
+from lunavl.sdk.descriptors.descriptors import (
+    BaseDescriptorBatch,
+    BaseDescriptor,
+    FaceDescriptorFactory,
+    HumanDescriptorFactory,
+)
+from lunavl.sdk.errors.errors import LunaVLError
+from lunavl.sdk.errors.exceptions import LunaSDKException
+from .base import CoreIndex
+from .stored_index import DynamicIndex, IndexType, DenseIndex
+
+
+class IndexBuilder(CoreIndex):
+    """
+    Index builder class
+
+    Attributes:
+        _bufSize (int): storage size with descriptors
+        _faceEngine (VLFaceEngine): faceEngine
+        _descriptorFactory (BaseDescriptorFactory): descriptor factory
+    """
+
+    def __init__(
+            self, faceEngine: PyIFaceEngine, descriptorFactory: Union[FaceDescriptorFactory, HumanDescriptorFactory]
+    ):
+        super().__init__(faceEngine.createIndexBuilder(), descriptorFactory)
+        self._bufSize = 0
+        self._faceEngine = faceEngine
+        self._descriptorFactory = descriptorFactory
+
+    @property
+    def bufSize(self) -> int:
+        """Get storage size with descriptors."""
+        return self._bufSize
+
+    def append(self, descriptor: BaseDescriptor) -> None:
+        """
+        Appends descriptor to internal storage.
+        Args:
+            descriptor: descriptor with correct length, version and data
+        Raises:
+            LunaSDKException: if an error occurs while adding the descriptor
+        """
+        error = self._coreIndex.appendDescriptor(descriptor.coreEstimation)
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        self._bufSize += 1
+
+    def appendBatch(self, descriptorsBatch: BaseDescriptorBatch) -> None:
+        """
+        Appends batch of descriptors to internal storage.
+        Args:
+            descriptorsBatch: Batch of descriptors with correct length, version and data
+        Raises:
+            LunaSDKException: if an error occurs while adding the batch of descriptors
+        """
+        error = self._coreIndex.appendBatch(descriptorsBatch.coreEstimation)
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        self._bufSize += len(descriptorsBatch)
+
+    def __delitem__(self, index: int):
+        """
+        Removes descriptor out of internal storage.
+        Args:
+            index: identification of descriptors position in internal storage
+        Raises:
+            IndexError: if index out of range
+            LunaSDKException: if an error occurs while remove descriptor failed
+        """
+        super().__delitem__(index=index)
+        self._bufSize -= 1
+
+    def loadIndex(self, path: str, indexType: IndexType) -> Union[DynamicIndex, DenseIndex]:
```

Comment:
i think that need add descriptor version

Reply:
added and opened fsdk issue